tqdm/contrib/itertools.py

```diff
@@ -29,7 +29,7 @@ def product(*iterables, **tqdm_kwargs):
             total *= i
```

Comment:
Apart from some minor differences (in the tests for example), I only found one difference that seems to matter. I believe that the influence of actually using the `repeat` parameter is not yet reflected in the `total` in your version. On my branch I do this to get a correct `total`:

```suggestion
            total *= i
        total = total ** repeat
```

Without this fix I believe the progress bar will exceed 100% (after which it just starts displaying the amount of iterations).